core_lib/helpers/logging.py

```diff
@@ -1,20 +1,21 @@
 import logging
 from functools import wraps
 
-from core_lib.helpers.func_utils import build_value_by_func_parameters, get_calling_module
+from core_lib.helpers.func_utils import build_value_by_func_parameters
 
 
 class Logging(object):
-    def __init__(self, message: str = '', level: int = logging.INFO, stack_depth=1):
+    def __init__(self, message: str = '', level: int = logging.INFO):
         self.message = message
         self.level = level
-        self.calling_module = get_calling_module(stack_depth)
 
     def __call__(self, func, *args, **kwargs):
         @wraps(func)
         def __wrapper(*args, **kwargs):
-            message = build_value_by_func_parameters(self.message, func, *args, **kwargs)
-            logging.getLogger(self.calling_module).log(self.level, '{}.{}'.format(self.message, ''.join(message)))
+            formatted_message = build_value_by_func_parameters(self.message, func, *args, **kwargs)
+            logging.getLogger(func.__qualname__).log(
+                self.level, f'{self.message if formatted_message == func.__qualname__ else formatted_message}'

```

Comment:
I was trying to implement it outside the logging too but the issue is, if a user is deliberately keeping the message empty then we will print the class name 2 times
`['INFO:TestLogging.logging_without_message:TestLogging.logging_without_message']`
something like this

Reply:
we are talking about a case where `self.message` ==  `func.__qualname__`
meaning the user pass to the message the function name?

Reply:
`logging.getLogger(func.__qualname__).log(
                self.level, f'{self.message if formatte...`

getLogger(func.__qualname__) is responsible for logging the class name and function, and it is being logged with every log, so by following your suggested code it will print the classname again, because the message is empty, `build_function_key` will also return the `func.__qualname__` and it will get printed twice

Reply:
Thank you for the explanation. yes considering that input. 
will this work?

```Python
formatted_message = build_value_by_func_parameters(self.message, func, *args, **kwargs) if self.message else ''
logging.getLogger(func.__qualname__).log(self.level, f'{formatted_message})'
```

Reply:
yes, thankyou